patches/third_party-blink-renderer-modules-storage-cached_storage_area.cc.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/third_party/blink/renderer/modules/storage/cached_storage_area.cc b/third_party/blink/renderer/modules/storage/cached_storage_area.cc
+index cb7bf9e97a0138904c0d9f9522e335f4e493d79a..557cf4c91f832d56d13f3a7b7cb635cc8cd029fe 100644
+--- a/third_party/blink/renderer/modules/storage/cached_storage_area.cc
++++ b/third_party/blink/renderer/modules/storage/cached_storage_area.cc
+@@ -244,7 +244,7 @@ void CachedStorageArea::BindStorageArea(
+     remote_area_.Bind(std::move(new_area), task_runner);
+   } else if (storage_namespace_) {
+     storage_namespace_->BindStorageArea(
```

Comment:
`#define BindStorageArea(STORAGE_KEY, LOCAL_FRAME_TOKEN, RECEIVER) BindStorageArea(STORAGE_KEY, IsSessionStorage() ? storage_key_ : local_dom_window->GetEphemeralStorageKeyOrStorageKey(), local_dom_window->GetLocalFrameToken(), RECEIVER)

Reply:
Can we do that? That's trying to replace this:

```
storage_namespace_->BindStorageArea(
    storage_key_, local_dom_window->GetLocalFrameToken(),
    remote_area_.BindNewPipeAndPassReceiver(task_runner));
```

but there are also two-parameter variants with that same function name in this scope, such as:

```
BindStorageArea(std::move(storage_area), local_dom_window);
```

Reply:
that shouldn't matter because it's defining a macro-like function with three parameters so it shouldn't match the two param version

Reply:
this is different than a regular macro matching `BindStorageArea` which would match both

Reply:
Yeah, I definitely see what you're saying, but the compiler seems not to like it:

```
In file included from ../../brave/chromium_src/third_party/blink/renderer/modules/storage/cached_storage_area.cc:32:
../../..\src/third_party/blink/renderer/modules/storage/cached_storage_area.cc(199,60): error: too few arguments provided to function-like macro invocation
  199 |   BindStorageArea(std::move(storage_area), local_dom_window);
      |                                                            ^
../../brave/chromium_src/third_party/blink/renderer/modules/storage/cached_storage_area.cc(25,9): note: macro 'BindStorageArea' defined here
   25 | #define BindStorageArea(STORAGE_KEY, LOCAL_FRAME_TOKEN, RECEIVER)   \
      |         ^
```

It's possible I'm missing something, though. I've asked @cdesouza-chromium to take a look to double-check me there.

Reply:
I think we'll go with what we have for now, but happy to address this in a follow-up if there's some better way forward. @cdesouza-chromium ran this through Compiler Explorer, though, and it seems like it's not supported: https://godbolt.org/z/jbrYxvfsE

Reply:
Everything here will be removed soon. Please don't waste time.

Reply:
@emerick I suppose it makes sense that it just tries to match all instances of `BindStorageArea` and then fails on the one that doesn't have the correct number of params. We're kind of abusing the preprocessor here so I guess I shouldn't expect it to handle this the way we would like.